code/mob/living/carbon/human.dm

```diff
@@ -3269,6 +3265,10 @@
 	if (src.juggling())
 		src.drop_juggle()
 
+/mob/living/carbon/human/set_a_intent(intent)
+	..()
+	if(src.equipped()?.item_function_flags & USE_INTENT_SWITCH_TRIGGER)
+		src.equipped().intent_switch_trigger(src)
```

Comment:
if these procs are exactly the same on both children types, why not move to the parent?

Reply:
Figured slight performance, don't know if it matters, since other mobs don't need to be checking for this

Reply:
I think it's safe to assume if a mob is calling `set_a_intent` then it wants intent things to work?

Reply:
Sure